src/main/java/core/basesyntax/service/impl/ShopServiceImpl.java

```diff
@@ -0,0 +1,78 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.dao.FruitTransactionDao;
+import core.basesyntax.model.Fruit;
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.ShopService;
+import core.basesyntax.service.quantity.CounterHandler;
+import core.basesyntax.strategy.CounterStrategy;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class ShopServiceImpl implements ShopService {
+    private static final int TYPE_INDEX = 0;
+    private static final int FRUIT_INDEX = 1;
+    private static final int QUANTITY_INDEX = 2;
+    private final FruitTransactionDao fruitTransactionDao;
+    private final CounterStrategy counterStrategy;
+
+    public ShopServiceImpl(FruitTransactionDao fruitTransactionDao,
+                           CounterStrategy counterStrategy) {
+        this.fruitTransactionDao = fruitTransactionDao;
+        this.counterStrategy = counterStrategy;
+    }
+    
+    @Override
+    public void processTheData(List<String[]> fileInfo) {
```

Comment:
this method responsible for parsing data from String[] to fruit transaction and adding it to Storage, it is violation of SPR (lets create separate class Parser for this purpouse)

Reply:
So I need to add fruits into storage manually in main method, using FruitDao? Did I get it correctly?